docs/proposals/scheduling/20220701-pod-migration-job.md

```diff
@@ -0,0 +1,393 @@
+	// Status represents the current status of PodMigrationJob
+	// e.g. ReservationCreated
+	Status string `json:"state,omitempty"`
+	// Reason represents a brief CamelCase message indicating details about why the PodMigrationJob is in this state.
+	Reason string `json:"reason,omitempty"`
+	// Message represents a human readable message indicating details about why the PodMigrationJob is in this state.
+	Message string `json:"message,omitempty"`
+	// Conditions records the stats of PodMigrationJob
+	Conditions []PodMigrationJobCondition `json:"conditions,omitempty"`
+	// NodeName represents the node's name of migrated Pod
+	NodeName string `json:"nodeName,omitempty"`
+	// PodsRef represents the newly created Pods after being migrated
+	PodsRef []v1.ObjectReference `json:"podsRef,omitempty"`
+	// PreemptedPodsRef represents the Pods that be preempted
+	PreemptedPodsRef []v1.ObjectReference `json:"preemptedPodsRef,omitempty"`
+	// PreemptedPodsReservations records information about Reservations created due to preemption
+	PreemptedPodsReservations []PodMigrationJobPreemptedReservation `json:"preemptedPodsReservation,omitempty"`
+}
+
+type PodMigrationJobPreemptedReservation struct {
+	// Namespace represents the namespace of Reservation
+	Namespace string `json:"namespace,omitempty"`
+	// Name represents the name of Reservation
+	Name string `json:"name,omitempty"`
+	// NodeName represents the assigned node for Reservation by scheduler
+	NodeName string `json:"nodeName,omitempty"`
+	// State represents the state of Reservation
+	State string `json:"state,omitempty"`
+	// PreemptedPodRef represents the Pods that be preempted
+	PreemptedPodRef *v1.ObjectReference `json:"preemptedPodRef,omitempty"`
+	// PodsRef represents the newly created Pods after being preempted
+	PodsRef []v1.ObjectReference `json:"podsRef,omitempty"`
+}
+
+type PodMigrationJobCondition struct {
+	// Type is the type of the condition.
+	Type PodMigrationJobConditionType `json:"type"`
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	Status PodMigrationJobConditionStatus `json:"status"`
+	// Last time we probed the condition.
+	// +nullable
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+	// Last time the condition transitioned from one status to another.
+	// +nullable
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human-readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+type PodMigrationJobPhase string
+
+const (
+	PodMigrationJobPending PodMigrationJobPhase = "Pending"
+	PodMigrationJobRunning PodMigrationJobPhase = "Running"
+	PodMigrationJobSucceed PodMigrationJobPhase = "Succeed"
+	PodMigrationJobFailed  PodMigrationJobPhase = "Failed"
+	PodMigrationJobAborted PodMigrationJobPhase = "Aborted"
```

Comment:
Could you please elaborate the usage differences between phase `Failed`, `Aborted`, `Unknown`?

Reply:
done